server/gc/src/test/java/org/apache/accumulo/gc/GarbageCollectionTest.java

```diff
@@ -644,4 +768,47 @@ public void bulkImportReplicationRecordsPreventDeletion() throws Exception {
     assertEquals(1, gce.deletes.size());
     assertEquals("hdfs://foo.com:6000/accumulo/tables/2/t-00002/A000002.rf", gce.deletes.get(0));
   }
+
+  /**
+   * Minimal test to show that dir and prevRow are required for valid scan
+   */
+  @Test
+  public void testDirAndPrevRow() throws Exception {
+
+    GarbageCollectionAlgorithm gca = new GarbageCollectionAlgorithm();
+
+    TestGCE gce = new TestGCE();
+    gce.candidates.add("/1636/default_tablet");
+    gce.addDirReference("1636", null, "/default_tablet");
+    gce.addPrevRowReference("1636", null);
+    gca.collect(gce);
+  }
+
+  /**
+   * Show that IllegalState is thrown when no dir entry present in metadata scan.
+   */
+  @Test
+  public void testNoDir() {
+
+    GarbageCollectionAlgorithm gca = new GarbageCollectionAlgorithm();
+
+    TestGCE gce = new TestGCE();
+    gce.candidates.add("/1636/default_tablet");
+    gce.addPrevRowReference("1636", null);
+    assertThrows(IllegalStateException.class, () -> gca.collect(gce));
+  }
+
+  /**
+   * Show that IllegalState is thrown when no prev row present in metadata scan.
+   */
+  @Test
+  public void testNoPrevRow() {
+
+    GarbageCollectionAlgorithm gca = new GarbageCollectionAlgorithm();
+
+    TestGCE gce = new TestGCE();
+    gce.candidates.add("/1636/default_tablet");
+    gce.addDirReference("1636", null, "/default_tablet");
+    assertThrows(IllegalStateException.class, () -> gca.collect(gce));
+  }
```

Comment:
Could add a third test for tablet row that has files, but no dir or prev row.

Reply:
Looking at this but my first attempts fail to detect when there are only file entries - I think it is that the row is not being updated - and could be an artifact of the test env.

Reply:
I think the tests added in 01e47c611b cover this?

Reply:
@keith-turner f98bb393ec tries to account for having at least one table / tablet scanned.  Not sure if this is the correct approach or a valid assumption.  For 1.x it may be (rep, and trace?), but not sure if there is a better way?

The idea was that if only ~del markers are seen and no rows, then probably best not to proceed.

And if there are no tablets - well there should not be much to gc anyway...

Reply:
I misinterpreted the suggestion.  b187550975 adds test for case where only prev row is present,  The checking for at least one tablet seen was not necessary and has been reverted with b187550975

Reply:
The following is what I was thinking about, a test where there is a file but no dir or prev row.

```java
  @Test
  public void testNoPrevRowNoDir() throws Exception {

    GarbageCollectionAlgorithm gca = new GarbageCollectionAlgorithm();

    TestGCE gce = new TestGCE();
    gce.candidates.add("/1636/default_tablet");
    gce.addFileReference("b", "m", "hdfs://foo.com:6000/user/foo/tables/b/t-0/F00.rf");
    assertThrows(IllegalStateException.class, () -> gca.collect(gce));
  }
```

Reply:
test added with 3bba72af51